internal/cmdutil/rand.go

```diff
@@ -0,0 +1,29 @@
+package cmdutil
+
+import (
+	"crypto/rand"
+	"fmt"
+	"math"
+	"math/big"
+	"os"
+	"strconv"
+)
+
+// NewRNGSeed generates a random seed value unless the env var OSBUILD_RNG_SEED
+// is set.
```

Comment:
Added but instead running all the tests in one function to force them to run sequentially and not have to mock the function.

Reply:
Thanks! This is fine of course. I'm slightly weary about this though because if we ever use `OSBUILD_RNG_SEED` inside any of our unit tests (e.g. to make results predictable for some specific thing in the future) there is a (small) racy that this test that sets the global env runs in parallel. Probably a small risk but wanted to mention it because if it happens it's a fun source of head scratching :) [but maybe I'm overthinking it, a common problem]

Reply:
I guess it's a tradeoff.  Not mocking `os.Getenv()` makes the test more "realistic" (for example, the way I wrote it now will catch failures if the env var name somehow changes and doesn't use the global const to do it) but you're right that it relies on controlling the environment variables for the whole test runtime which could affect other tests.

Reply:
Should anyway use `t.Setenv()` instead.

Reply:
> does not help with the fundamental problem, it seems it does not serilaize tests that touch the same env (which it probably could and that would be super nice)

Yeah I was hoping it would do that.  Maybe as an alternative it could catch calls to the function that happen in parallel tests, but it doesn't do that either.

Reply:
Let's not block on this, we can change this in the future if it becomes a problem :)